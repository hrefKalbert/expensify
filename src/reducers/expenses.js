// Expense Reducer

const expensesReducerDefaultState = []

const expensesReducer = (state = expensesReducerDefaultState, action) => {
  switch (action.type) {
    case 'ADD_EXPENSE':
      return [
        ...state, action.expense
      ]
    case 'REMOVE_EXPENSE':
      return state.filter(({ id }) => id !== action.id)
    case 'EDIT_EXPENSE':
      return state.map(cur => {
        if (cur.id === action.id) {

          return {
            ...cur,
            ...action.updates

          }

        } else {
          return cur;
        }
      })
    default:
      return state
  }
}

export default expensesReducer