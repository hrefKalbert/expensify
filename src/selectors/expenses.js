import moment from 'moment'

export const getVisibleExpenses = (expenses, { text, sortBy, startDate, endDate }) => {
  return expenses.filter((cur)=> {
    const momentCreatedAt = moment(cur.createdAt)
    const startDateMatch = startDate ? startDate.isSameOrBefore(momentCreatedAt, 'day') : true
    const endDateMatch = endDate ? endDate.isSameOrAfter(momentCreatedAt, 'day') : true
    const textMatch = cur.description.toLowerCase().includes(text.toLowerCase())
  
    return startDateMatch && endDateMatch && textMatch
  }).sort((a,b) => {
    if(sortBy === 'date') {
      return a.createdAt < b.createdAt ? 1  : -1
    } else if(sortBy === 'amount'){
      return a.amount < b.amount ? 1 : -1
    }
  })
}