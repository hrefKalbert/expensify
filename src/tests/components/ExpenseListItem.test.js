import React from 'react'
import ExpenseListItems from '../../components/ExpenseListItem'
import {shallow} from 'enzyme'
import expenses from '../fixtures/expenses'

test('should Render ExpenseListItem with fixture data',() => {
  const wrapper = shallow(<ExpenseListItems {...expenses[0]}/>)
  expect(wrapper).toMatchSnapshot()
  // console.log('Check the data:', expenses)
})