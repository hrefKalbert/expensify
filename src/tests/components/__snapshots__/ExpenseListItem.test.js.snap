// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should Render ExpenseListItem with fixture data 1`] = `
<div>
  <Link
    to="/edit/1"
  >
    <h1>
      Gum
    </h1>
  </Link>
  <p>
    Amount: 
    $1.95
  </p>
  <p>
    Created at: 
    January 1st, 1970
  </p>
</div>
`;
